### PR TITLE
[TrafficControl][Cherry Pick] Handle x-forwarded-for with and without port

### PR DESCRIPTION
**Pull request by williampsmith**
## Description 

Describe the changes or additions included in this PR.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

**Review by chris-gorham — APPROVED** [review]
lgtm

**Review by chris-gorham — APPROVED** [review]
LGTM

File: crates/sui-core/src/authority_server.rs [hunk 1]
```diff
@@ -999,20 +999,18 @@ macro_rules! handle_with_decoration {
                 let do_header_parse = |op: &MetadataValue<Ascii>| {
                     match op.to_str() {
                         Ok(header_val) => {
-                            match header_val.parse::<SocketAddr>() {
-                                Ok(socket_addr) => Some(socket_addr.ip()),
-                                Err(err) => {
+                            header_val.parse::<IpAddr>().ok().or_else(|| {
+                                header_val.parse::<SocketAddr>().ok().map(|socket_addr| socket_addr.ip()).or_else(|| {
```
Comment (mystenmark) [inline crates/sui-core/src/authority_server.rs:1003]: might be worth pulling this code out into a function, both for clarity and for re-use with axum_router

**Review by mystenmark — APPROVED** [review]

**Comment by williampsmith** [comment]
Rolled into https://github.com/MystenLabs/sui/pull/18699 and already rolled out

**Pull request closed without merging**